parser/parser.go

```diff
@@ -0,0 +1,24 @@
+package parser
+
+import (
+	"github.com/martin-helmich/prometheus-nginxlog-exporter/config"
+	"github.com/martin-helmich/prometheus-nginxlog-exporter/parser/jsonparser"
+	"github.com/martin-helmich/prometheus-nginxlog-exporter/parser/textparser"
+)
+
+// Parser parses a line of log to a map[string]string.
+type Parser interface {
+	ParseString(line string) (map[string]string, error)
+}
```

Comment:
I think I mentioned it earlier already, but I really like this refactoring. 🤩  It comes with minimal changes to the original code, but adds so much extensibility. Well done! :+1:

Reply:
You inspired me.🤝😊